smiles/views.py

```diff
@@ -220,11 +226,86 @@ def upload_smile_dps(request, dp_type):
     if old_schema_param == 'true':
         smiles_data_migration.migrate_smiles_dps(request_data, file_path, dp_type.value)
     else:
-        smiles_dp_util.upload_smiles_data_products.delay(request_data, file_path, dp_type.value)
+        smiles_dp_util.upload_smiles_data_products.delay(request_data, file_path, dp_type.value) # removed the '.delay' just for the sake of debugging
 
 
 def extract_request_data(request):
     return {
         'user_id': str(request.user.id),
         'tenant_id': "demotenant"
     }
+###################
+@csrf_exempt
+def unified_smiles_action(request):
+    
+    action_type = request.GET.get('action', '').strip()  # get action: 'search' or 'share'?
+
+    if action_type == 'search':
+        return search_users_and_groups_logic(request)
+    elif action_type == 'share':
+        return share_data_products_logic(request)
+    else:
+        return JsonResponse({'error': 'Invalid action type'}, status=400)
+
+
+def search_users_and_groups_logic(request):
+    search_term = request.GET.get("query", "").strip()  
+    tenant_id = request.GET.get("tenant_id", "").strip()
+    
+
+    if not tenant_id:
+        tenant_id = "demotenant"  # if no tenant_id，assisn to "demotenant"
+       
+
+    results = {"users": [], "groups": []}
+    try:
+        # 
+        with grpc.insecure_channel('localhost:6565') as channel:
```

Comment:
better to make this host and port be extracted from configurations rather than hardcoding. 
If you're doing this change make sure to update the `data_catalog_service` as well